carnot/carnot_vote_aggregation.py

```diff
@@ -0,0 +1,421 @@
+# Carnot-2 is extension of the Carnot protocol. Carnot-2 is designed to include majority of votes in the QC as a proof.
+# Since aggregating signatures is expensive, therefore Carnot-2 has been designed to optimize signature aggregation
+# Below is the description of the Carnot-2 Protocol.
+# Happy Path:
+
+# Step 1: Vote Multicast
+# Associated Function: Send(to=recipient, payload=vote)
+# Description: Each node multicasts its vote to the members of its committee.
+# Step 2: Certificate Generation
+# Associated Function: approve_block(block: Block, votes: Set[Vote]) -> Event, build_qc(self, view: View, block: Optional[Block], timeouts: Optional[Set[Timeout]]) -> Qc
+# Description: Each node generates a certificate by collecting votes from at least 2/3 of the members in its committee.
+# Step 3: Certificate Transmission
+# Associated Function: forward_vote_qc(self, vote: Optional[Vote] = None, qc: Optional[Qc] = None) -> Optional[Event]:
+# Description: Forward a QC if it is built by the timeout t1 else forward votes.
```

Comment:
is it a typo that votes is plural in the description but only one is provided in the function argument?